app/controllers/chapters_controller.rb

```diff
@@ -93,8 +93,12 @@ def edit
     if params["remove"] == "me"
       @chapter.authors_to_remove = current_user.pseuds
       @chapter.save
-      flash[:notice] = ts("You have been removed as a creator from the chapter")
-      redirect_to @work
+      if @work.chapters.any? { |c| current_user.is_author_of?(c) }
+        flash[:notice] = ts("You have been removed as a creator from the chapter")
+        redirect_to @work
+      else # remove from work if no longer co-creator on any chapter
+        redirect_to controller: "works", action: "edit", id: @work.id, remove: "me"
```

Comment:
Is it possible to do this as `redirect_to edit_work_path(@work, remove: "me")` instead? (I haven't actually done a full review, so feel free to let this wait in case something else comes up.)

Reply:
I'll give it a try :+1: I've been meaning to look into how rails ~~magics~~ routes things like 'edit_work_path'